crates/xxi-node/src/message_handler.rs

```diff
@@ -465,46 +476,75 @@ impl TenTenOneMessage {
     /// an offer so if an offer is passed the function will panic. This is most likely not a future
     /// proof solution as we'd might want to enrich the response with 10101 metadata as well. If
     /// that happens we will have to rework this part.
-    pub fn build_from_response(message: Message, is_rollover: bool) -> Self {
+    pub fn build_from_response(message: Message, order_id: Option<Uuid>) -> Self {
         match message {
             Message::Channel(ChannelMessage::Accept(accept_channel)) => {
-                TenTenOneMessage::Accept(TenTenOneAcceptChannel { accept_channel })
+                TenTenOneMessage::Accept(TenTenOneAcceptChannel {
+                    accept_channel,
+                    order_id: order_id.expect("to be some"),
+                })
             }
             Message::Channel(ChannelMessage::Sign(sign_channel)) => {
-                TenTenOneMessage::Sign(TenTenOneSignChannel { sign_channel })
+                TenTenOneMessage::Sign(TenTenOneSignChannel {
+                    sign_channel,
+                    order_id: order_id.expect("to be some"),
+                })
             }
             Message::Channel(ChannelMessage::SettleAccept(settle_accept)) => {
-                TenTenOneMessage::SettleAccept(TenTenOneSettleAccept { settle_accept })
+                TenTenOneMessage::SettleAccept(TenTenOneSettleAccept {
+                    settle_accept,
+                    order_id: order_id.expect("to be some"),
+                })
             }
             Message::Channel(ChannelMessage::SettleConfirm(settle_confirm)) => {
-                TenTenOneMessage::SettleConfirm(TenTenOneSettleConfirm { settle_confirm })
+                TenTenOneMessage::SettleConfirm(TenTenOneSettleConfirm {
+                    settle_confirm,
+                    order_id: order_id.expect("to be some"),
+                })
             }
             Message::Channel(ChannelMessage::SettleFinalize(settle_finalize)) => {
-                TenTenOneMessage::SettleFinalize(TenTenOneSettleFinalize { settle_finalize })
+                TenTenOneMessage::SettleFinalize(TenTenOneSettleFinalize {
+                    settle_finalize,
+                    order_id: order_id.expect("to be some"),
```

Comment:
🔧 I don't love these `expect`s. We could make the function fallible since it's called in a fallible context, and also pattern match against the `order_id` to avoid the `expect`s and the `is_none`s.